setup/includes/upgrades/common/3.0.0-update-sys-setting_upload_files.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Common upgrade script for modify upload_files System Setting
+ *
+ * @var modX $modx
+ * @package setup
+ */
+
+use MODX\Revolution\modSystemSetting;
+
+$messageTemplate = '<p class="%s">%s</p>';
+
+$keysRemove = ['upload_images', 'upload_media'];
+
+/** @var modSystemSetting $setting */
+$upload_files_setting = $modx->getObject(modSystemSetting::class, ['key' => 'upload_files']);
+
+// Update upload_files
+if ($upload_files_setting) {
+    $success = false;
+
+    $upload_files_value = $upload_files_setting->get('value');
+    $upload_files_arr = explode(',', $upload_files_value);
+
+    // Adding webp, woff2
+    if (!in_array('webp', $upload_files_arr) || !in_array('woff2', $upload_files_arr)) {
+        $upload_files_setting->set('value', $upload_files_value . ',webp,woff2');
+        $upload_files_setting->save();
+    }
+
+    // Collecting all extensions in upload_files, upload_images, upload_media
+    $upload_images_setting = $modx->getObject(modSystemSetting::class, ['key' => 'upload_images']);
+    $upload_media_setting = $modx->getObject(modSystemSetting::class, ['key' => 'upload_media']);
+
+    $upload_images_arr = $upload_media_arr = [];
+    if ($upload_images_setting) {
+        $upload_images_value = $upload_images_setting->get('value');
+        $upload_images_arr = explode(',', $upload_images_value);
+    }
+
+    if ($upload_media_setting) {
+        $upload_media_value = $upload_media_setting->get('value');
+        $upload_media_arr = explode(',', $upload_media_value);
+    }
+
+    $upload_files_arr_new = array_unique(array_merge($upload_files_arr, $upload_images_arr, $upload_media_arr));
+    $upload_files_arr_new = array_filter($upload_files_arr_new, fn($value) => !is_null($value) && $value !== '');
```

Comment:
[Arrow shorthand functions](https://www.php.net/manual/en/functions.arrow.php) were introduced in php 7.4, officially MODX3 still supports 7.2.

(Bumping that is a good topic to discuss elsewhere.)

Reply:
It also looks like this particular callback could simply be removed as the default filter array_filter does will already filter out empty values.